test/performance/alignment/global_affine_alignment_simd_benchmark.cpp

```diff
@@ -0,0 +1,183 @@
+// -----------------------------------------------------------------------------------------------------
+
+#include <benchmark/benchmark.h>
+
+#include <utility>
+#include <vector>
+
+#include <seqan3/alignment/pairwise/align_pairwise.hpp>
+#include <seqan3/alphabet/nucleotide/dna4.hpp>
+#include <seqan3/range/views/zip.hpp>
+#include <seqan3/std/ranges>
+#include <seqan3/test/performance/sequence_generator.hpp>
+#include <seqan3/test/performance/units.hpp>
+#include <seqan3/test/seqan2.hpp>
+
+#ifdef SEQAN3_HAS_SEQAN2
+    #include <seqan/align.h>
+    #include <seqan/align_parallel.h>
+#endif
+
+constexpr auto affine_cfg = seqan3::align_cfg::mode{seqan3::global_alignment} |
+                            seqan3::align_cfg::gap{seqan3::gap_scheme{seqan3::gap_score{-1},
+                                                                      seqan3::gap_open_score{-10}}} |
+                            seqan3::align_cfg::scoring{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
+                                                                                         seqan3::mismatch_score{-5}}};
+
+// Aliases to beautify the benchmark output
+using score = seqan3::detail::with_score_type;
+using end_position = seqan3::detail::with_back_coordinate_type;
+using trace = seqan3::detail::with_alignment_type;
+
+// Globally defined constants to ensure same test data.
+inline constexpr size_t sequence_length_mean = 150;
+inline constexpr size_t set_size        = 1024;
+
+// Range to test for sequence length variance
+inline constexpr size_t deviation_begin = 0;
+inline constexpr size_t deviation_end = 64;
+inline constexpr size_t deviation_step = 8;
+
+//We don't know if the system supports hyper threading so we use only halve the threads so that the
+// simd benchmark is likely to run on physical cores only.
+uint32_t get_number_of_threads()
+{
+    uint32_t thread_count = std::thread::hardware_concurrency();
+    return (thread_count == 1) ? thread_count : thread_count >> 1;
+}
+
+// ============================================================================
+//  affine; score; dna4; collection
+// ============================================================================
+
+template <typename result_t>
+void seqan3_affine_dna4_simd(benchmark::State & state)
+{
+    size_t sequence_length_variance = state.range(0);
+    auto data = seqan3::test::generate_sequence_pairs<seqan3::dna4>(sequence_length_mean,
+                                                                    set_size,
+                                                                    sequence_length_variance);
+
+    int64_t total = 0;
+    auto vectorised_config = affine_cfg |
+                             seqan3::align_cfg::result{result_t{}, seqan3::using_score_type<int16_t>} |
+                             seqan3::align_cfg::vectorise;
+    for (auto _ : state)
+    {
+        for (auto && res : seqan3::align_pairwise(data, vectorised_config))
```

Comment:
This might lead to numbers that are not comparable anymore if the architecture is fast enough to compute all the data within the testing scope of google benchmark (was it 1 sec?), or not?
You could maybe circumvent this by adding a `while (true)` around it.

Reply:
>This might lead to numbers that are not comparable anymore if the architecture is fast enough to compute all the data within the testing scope of google benchmark (was it 1 sec?), or not?

Is that so? I thought this is all handled by benchmark. They first make a few runs and determine the number of iterations based on that. I mean it can handle functions in ns area, so I think that should be fine.

Reply:
Sorry, I confused this with a situation in the file formats where there were spurious errors because the file was empty.